util/preseeds.robot

```
@@ -0,0 +1,135 @@
+# TODO: maybe have a single file to include if we need to include the same
+# stuff in all test cases
+Resource        ../variables.robot
+Resource        ../keywords.robot
+Resource        ../keys.robot
+Resource        ../keys-and-keywords/ubuntu-keywords.robot
+
+Suite Setup     Run Keywords
+...                 Prepare Test Suite
+# Test suite used to quickly flash disk images with preinstalled OSes.
+# Two environment variables must be set prior to running it:
+# - SOURCE_IMAGE
+# - TARGET_DISK
+# Optional:
+# - CLONEZILLA_TTY
+#
+# Example usage:
+# - Flash the `1_windows_ubuntu` image onto the `nvme0n1` disk on the device using ${RTE_IP}:
+#    SOURCE_IMAGE=1_windows_ubuntu TARGET_DISK=nvme0n1 ./scripts/run.sh util/preseeds.robot -- -t "Restore Disk*"
+# - The same, but redirect the Clonezilla's output onto ttyS0 if video output is not available:
+#    SOURCE_IMAGE=1_windows_ubuntu TARGET_DISK=nvme0n1 CLONEZILLA_TTY=ttyS0 ./scripts/run.sh util/preseeds.robot -- -t "Restore Disk*"
+# - Launch clonezilla in manual mode over chosen serial console (for restore or upload):
+#    CLONEZILLA_TTY=ttyUSB0 ./scripts/run.sh util/preseeds.robot -- -t "Manual*"
+#
+#    Important: Don't run the whole test suite. Every test case is a
+#    separate functionality and running them all after each other does not
+#    make much sense, unless you know what you are doing.
+
+
+*** Variables ***
+${CLONEZILLA_IPXE_SERVER}=      http://192.168.10.217:8080
+${DISKS_NFS_IP}=                192.168.10.217
+${DISKS_NFS_PATH}=              /srv/nfs/disk-images
+${TIME_LIMIT}=                  40m
+
+
+*** Test Cases ***
+Restore Disk Clonezilla
+    ${clonezilla_tty}=    Get Envvar    CLONEZILLA_TTY    ${TRUE}
+    ${source_image}=    Get Envvar    SOURCE_IMAGE
+    ${target_disk}=    Get Envvar    TARGET_DISK
+
+    Power On
+    ${ipxe_entered}=    Run Keyword And Return Status    Enter IPXE
+    IF    not ${ipxe_entered}    # It might just be disabled
```

We also have `Make Sure That Network Boot Is Enabled` - maybe it could be used here prior entering iPXE?

---

All `Make Sure That Network Boot Is Enabled` does is just call `Set UEFI Option`, and it fails silently instead of Failing if it is not possible, so it's not equivalent.

I think a better alternative would be to extend `Enter IPXE` a bit so that it performs this check and optionally enables Network Boot if disabled.

---

Let's do it or at least note it down and link here.

---

https://github.com/Dasharo/open-source-firmware-validation/pull/901/commits/b37f7ba4b350e494e2037363cf6b8f948f530ea7
The change shouldn't collide with existing usages of this keyword, as it only does something different if entering the IPXE menu fails